packages/paste-core/components/disclosure/__tests__/disclosure.test.tsx

```diff
@@ -18,6 +26,29 @@ export const MockDisclosure: React.FC<{
   );
 };
 
+const useVisibleDisclosureState = (): DisclosureStateReturn => {
+  const disclosure = useDisclosureState({visible: true});
+  return {
+    ...disclosure,
+    toggle: () => {
+      disclosure.toggle();
```

Comment:
This kind of doesn't test anything...
```
toggle: () => {disclosure.toggle();}
```
is equivalent to
```
toggle: disclosure.toggle
```
which does nothing.  Might as well return `disclosure` directly :P



Something like this might be better:
```js
const useVisibleDisclosureState = (): DisclosureStateReturn => {
  const [visible, setVisible] = React.useState(true);
  const toggle = () => {
    setVisible(!visible)
  };

  const disclosure = useDisclosureState({visible});
  return {
    ...disclosure,
    visible,
    toggle
  }
}
```
The above clearly shows that we now control the visible state, and we pass the value down to Disclosure to render. (I didn't test this code btw, hope it works 🤞)

Reply:
I've restructured this function as suggested. Just had to change it slightly to get it working.